src/components/clusterWizard/wizardTransition.ts

```diff
@@ -8,14 +15,111 @@ export type ClusterWizardStepsType =
 
 export const CLUSTER_WIZARD_FIRST_STEP: ClusterWizardStepsType = 'cluster-details';
 
-// We are collocating all these canNext* functions for easier maintenance.
-// However they should be independent on each other anyway.
 type TransitionBackendProps = { cluster: Cluster };
 type TransitionProps = TransitionBackendProps & { isValid?: boolean; isSubmitting?: boolean };
 
+const checkClusterValidations = (
+  clusterValidationsInfo: ClusterValidationsInfo,
+  group: 'hostsData' | 'network' | 'configuration',
+  requiredIds: ClusterValidationId[],
+): boolean =>
+  requiredIds.every((id) => {
+    const validation = clusterValidationsInfo[group].find((validation) => id === validation.id);
+    return validation?.status === 'success';
+  });
+
+const checkClusterValidationGroups = (
+  clusterValidationsInfo: ClusterValidationsInfo,
+  groups: ('hostsData' | 'network' | 'configuration')[],
+) =>
+  groups.every((group) =>
+    clusterValidationsInfo[group]?.every((validation) => validation.status === 'success'),
+  );
+
+const checkHostValidations = (
+  hostValidationsInfo: HostValidationsInfo,
+  group: 'hardware' | 'network',
+  requiredIds: HostValidationId[],
+): boolean =>
+  requiredIds.every((id) => {
+    const validation = hostValidationsInfo[group]?.find((validation) => id === validation.id);
+    return validation?.status === 'success';
+  });
```

Comment:
Following up on my previous comment, my intention was to remove the requirement to specify the group to allow running this function only once when we need to check validations picked from several groups which can happen on subsequent steps.

We are already flattening the validations info on some places:
https://github.com/openshift-assisted/assisted-ui-lib/blob/master/src/components/clusterConfiguration/ClusterValidationSection.tsx#L48

So for our case it would be:
```suggestion
  {
    const requiredValidations = _.values(hostValidationsInfo).flat().filter(v => requiredIds.includes(v.id))
    return requiredValidations.every(v => v.status === 'success);
    }
```

Reply:
ok, done.
Adding an extra check for the case that the input does not contain the required validation at all.